test/cql-pytest/test_materialized_view.py

```diff
@@ -1189,3 +1192,63 @@ def test_base_partition_deletion_with_low_timestamp(cql, test_keyspace):
+        matches = pattern.search(create_stmt)
+        match = matches.group(1)
+        assert match is not None
+        return match
+
+    with new_test_table(cql, test_keyspace, 'p int PRIMARY KEY') as table:
+        # Case 1. Try to create an MV with the ID of another MV.
+        with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL') as mv:
+            mv_id = get_id_of_cf(mv)
+            with pytest.raises(InvalidRequest, match=f'Table with ID {mv_id} already exists: {mv}'):
+                with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {mv_id}'):
+                    pass
+
+        # Case 2. Try to creat an MV with the ID of a regular table.
+        cf_id = get_id_of_cf(table)
+        with pytest.raises(InvalidRequest, match=f'Table with ID {cf_id} already exists: {table}'):
+            with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {cf_id}'):
+                pass
+
+# Verify that creating an MV with a specified ID suceeds and the MV really bears it.
+def test_creating_mv_with_given_id(cql, test_keyspace):
+    # Returns the ID of a given CF. `cf_full_name` is of form `ks_name.cf_name`.
+    def get_id_of_cf(cf_full_name: str) -> str:
+        create_stmt = cql.execute(f'DESC TABLE {cf_full_name} WITH INTERNALS').one().create_statement
+        pattern = re.compile(r'[WITH|AND] ID = (\S+)')
+        matches = pattern.search(create_stmt)
+        match = matches.group(1)
+        assert match is not None
+        return match
+
+    with new_test_table(cql, test_keyspace, 'p int PRIMARY KEY') as table:
+        # Step 1. We need an unused ID for the MV. Create a materialized view, obtain a statement
+        #         to restore it, and extract its ID. Drop the MV.
+        with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL') as mv:
+            mv_id = get_id_of_cf(mv)
+
+        # Step 2. Create a new MV with a specific ID. Verify that it really uses it.
+        with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {mv_id}') as mv:
+            assert mv_id == get_id_of_cf(mv)
+
+# Verify that creating an MV fails when provided an invalid ID.
+def test_creating_mv_with_invalid_id(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int PRIMARY KEY') as table:
+        def try_creating_mv_and_fail(mv_id: str, exception_type: Any, exception_match: str):
+            with pytest.raises(exception_type, match=exception_match):
+                with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {mv_id}'):
+                    pass
+
+        try_creating_mv_and_fail('null', ConfigurationException, 'Invalid table id')
+        try_creating_mv_and_fail("'null'", ConfigurationException, 'Invalid table id')
+        try_creating_mv_and_fail("'something'", ConfigurationException, 'Invalid table id')
+        try_creating_mv_and_fail('!@?', SyntaxException, 'Syntax error')
```

Comment:
This specific line isn't useful - it's indeed a syntax error, like it told you :-) You just create a statement that says "WITH ID = !@?" but that's a syntax error because it's not valid CQL, it's supposed to be a string in single quotes.

I'd simplify this test to just be, shorter and no more utility functions:

```python
for id in ["null", "'null'", "'something'", "'!@?'"]:
     with pytest.raises(ConfigurationException, match='Invalid table id'):
          with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {id}'):
                    pass
```

Again, please make sure this test also passes on Cassandra (I don't know if the string "Invalid table id" is the same there.

Reply:
This looks much better indeed :)